src/main/java/com/exadel/discountwebapp/history/service/HistoryService.java

```diff
@@ -0,0 +1,43 @@
+package com.exadel.discountwebapp.history.service;
+
+import com.exadel.discountwebapp.history.enums.OperationsEnum;
+import com.exadel.discountwebapp.history.mapper.HistoryMapper;
+import com.exadel.discountwebapp.history.vo.HistoryRequestVO;
+import com.exadel.discountwebapp.history.vo.HistoryResponseVO;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import org.javers.core.Javers;
+import org.javers.repository.jql.QueryBuilder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class HistoryService {
+    private final Javers javers;
+    private final HistoryMapper historyMapper;
+
+    @SneakyThrows
+    public List<HistoryResponseVO> getHistory(HistoryRequestVO request) {
```

Comment:
this method should not iterate over all OperationEnum values
it should take operationType from request

Reply:
changed